File: src/1_100/0007_Reverse_Integer/Reverse_Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if(x==0):
            return 0
        b=list(str(x))
        c=b[::-1]
        index=0
        minus_sign=len(c)-1
        has_minus=False
        for i in range(len(c)):
            if c[i]!='0':
                break
            index+=1
        if c[len(c)-1]=='-':
            minus_sign-=1
            has_minus=True
        result = int( "".join(c[index:minus_sign+1]) )
        if has_minus:
            result = 0 - result
        if (result<(-2**31) or result>2**31-1):
            result = 0
        return result
		
		
		
	def reverse(self, x):
		arr = list(str(x))
		is_negative = False
		start = 0
		n = len(arr)
		if arr[0] == '-':
			is_negative = True
			start = 1
		i = start
		j = n-1
		while i<j:
			arr[i],arr[j] = arr[j],arr[i]
			i += 1
			j -= 1
		res = int("".join(arr))
		if res<-2**31 or res>(2**31-1):
			return 0
		return res
		
		
		
	def reverse(self, x):
		if x<0:
			x = x*-1
			x = int("-"+str(x)[::-1])
		else:
			x = int(str(x)[::-1])
		if x>(2**31-1) or x<-2**31:
			return 0
		return x

File: src/1_100/0007_Reverse_Integer/Reverse_Integer.java
class Solution {
    public int reverse(int x) {
		if(x==0) {
			return 0;
		}
        long num = 0L;
		boolean is_negative = false;
		if(x < 0) {
			is_negative = true;
			x = -x;
		}
		//将 12345 转换为 54321，每次取模得到末数，再加上 num*10
		//因为num第一次是0，会自动忽略边界问题
		//如果x是120，第一次计算就是  num*10-> 0*10,  x%10->0, 0+0=0，自动忽略了边界问题
		while(x > 0) {
			num = num*10 + x%10;
			x /= 10;
		}
		if(is_negative) {
			return (int)(-num<Integer.MIN_VALUE? 0 : -num);
		}
		return (int)(num>Integer.MAX_VALUE? 0 : num);
    }
}